squad/core/comparison.py

```diff
@@ -54,6 +54,49 @@ def __init__(self, *builds):
         for build in self.builds:
             self.environments[build] = set()
 
+        if len(self.builds) >= 2 and None not in self.builds:
+            baseline = self.builds[0]
+            target = self.builds[1]
+
+            # This is ugly, but it's an easy, light-weight way to compare tests
+            # from two builds instead of loading all tests in memory.
+            # Please use Django ORM whenever it starts supporting
+            # queryies using the same table
+            self.base_sql = {
+                'select': [
+                    'target.id',
+                    'target.build_id',
+                    'target.environment_id',
+                    'target.test_run_id',
+                    'target.suite_id',
+                    'target.metadata_id',
+                ],
+                'from': [],
+                'where': [
+                    'baseline.build_id = {baseline_id}',
+                    'target.build_id = {target_id}',
+                    'baseline.metadata_id = target.metadata_id',
+                    'target.result IS NOT NULL',
+                    'baseline.result IS NOT NULL',
+                    'target.result != baseline.result',
+                ],
+                'values': {
+                    'baseline_id': baseline.id,
+                    'target_id': target.id
+                }
+            }
+
+            # If builds belong to different projects, environment comparison should
+            # use slug
+            if self.__same_projects__():
+                self.base_sql['where'].append('target.environment_id = baseline.environment_id')
```

Comment:
`from` seems like it would be empty here. Would that be a problem?

Reply:
That is expected. MetricComparison will fill that up with `core_metric` whereas TestComparison will do the same with `core_test`. The comparison approach is 99% the same for both test and metrics, except that each type need to use their own table names. Good observation tho!